jenkins/opensearch-ruby/Jenkinsfile

```diff
@@ -0,0 +1,66 @@
+pipeline {
+    options {
+        timeout(time: 1, unit: 'HOURS')
+    }
+    agent none
+    parameters {
+        string(
+                name: 'RELEASE_BRANCH',
+                description: 'Branch name to release',
+                trim: true
+        )
+    }
+    stages {
+        stage('ruby-build-sign-upload') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+                    image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211130'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    if (params.RELEASE_BRANCH == '') {
+                        currentBuild.result = 'ABORTED'
+                        error('RELEASE_BRANCH parameter is not set')
+                    }
+                    git url: 'https://github.com/opensearch-project/opensearch-ruby.git', branch: "$RELEASE_BRANCH"
+                    withCredentials([string(credentialsId: 'opensearchproject-rubygems-private-key', variable: 'RUBYGEMS_PRIVATE_KEY')]) {
+                        writeFile file: 'private_key.pem', text: RUBYGEMS_PRIVATE_KEY
+                    }
+                    sh'''
+                        keypath=$(realpath private_key.pem)
+                        for f in $(find . -name '*.gemspec');\\
+                            do echo $f;\\
+                            sed -i '$s#end#  s.signing_key = File.expand_path("'"$keypath"'") if $0 =~ /gem\\\\z/\\n&#' $f;\\
+                        done
+                        gem cert --add certs/opensearch-rubygems.pem
+                        bash release/build.sh
+                        for f in $(find dist/ -name '*.gem');\\
+                            do echo $f; gem install $f;\\
+                            val=$(echo $f | grep -Eo '[0-9]+\\.[0-9]+\\.[0-9]+');\\
+                            gem uninstall $f -v $val; gem install $f -P HighSecurity;\\
+                        done
+                        rm -f $keypath
```

Comment:
Don't need to clean up after the script, docker image gets erased, please remove to keep this as simple as possible.

Reply:
I added this on the @peterzhuamazon recommendation as he mentioned cleanup can get stuck and not remove the key. It's basically prevention is better than cure here.